### PR TITLE
Various Add-ons: More example alerts

### PR DESCRIPTION
**Pull request by kingthorin**
## Overview
- Scan rules > Add example alert functionality and alert refs where logical.
- Unit Tests > Add tests to assert the example alerts and references.
- CHANGELOG > Add notes.

## Related Issues
- zaproxy/zaproxy#6119
- zaproxy/zaproxy#7100
- zaproxy/zaproxy#8262

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [x] Write tests
- [x] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

### CODE REVIEW

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SourceCodeDisclosureWebInfScanRule.java [hunk 1]
```diff
@@ -344,4 +346,9 @@ public int getCweId() {
     public int getWascId() {
         return 34; // Predictable Resource Location
     }
+
+    @Override
+    public List<Alert> getExampleAlerts() {
+        return List.of(buildWebinfAlert("class A\n{\n}\n").build());
+    }
```
Comment (kingthorin) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SourceCodeDisclosureWebInfScanRule.java:353]: I guess I should alert ref this one and include both?

File: addOns/pscanrules/src/main/resources/org/zaproxy/zap/extension/pscanrules/resources/Messages.properties [hunk 2]
```diff
@@ -209,13 +209,13 @@ pscanrules.insecureauthentication.refs = https://cheatsheetseries.owasp.org/chea
...
+pscanrules.insecureformload.extrainfo = The response to the following request over HTTP included an HTTPS form tag action attribute value:\r\n\r\n{0}\r\nThe context was:\r\n\r\n{1}
 pscanrules.insecureformload.name = HTTP to HTTPS Insecure Transition in Form Post
 pscanrules.insecureformload.refs = 
 pscanrules.insecureformload.soln = Use HTTPS for landing pages that host secure forms.
 
 pscanrules.insecureformpost.desc = This check identifies secure HTTPS pages that host insecure HTTP forms. The issue is that a secure page is transitioning to an insecure page when data is uploaded through a form. The user may think they're submitting data to a secure page when in fact they are not.
-pscanrules.insecureformpost.extrainfo = The response to the following request over HTTPS included an HTTP form tag action attribute value:\r\n\r\n{0}The context was:\r\n\r\n{1}
+pscanrules.insecureformpost.extrainfo = The response to the following request over HTTPS included an HTTP form tag action attribute value:\r\n\r\n{0}\r\nThe context was:\r\n\r\n{1}
```
Comment (kingthorin) [inline addOns/pscanrules/src/main/resources/org/zaproxy/zap/extension/pscanrules/resources/Messages.properties:218]: For a separate PR should this simply use \n ?
Should that be enforced by the formatter?

**Comment by kingthorin** [comment]
Deconflicted.

**Comment by kingthorin** [comment]
Deconflicted, again

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SourceCodeDisclosureWebInfScanRule.java (hunk 1 above)
Comment (thc202) [reply addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/SourceCodeDisclosureWebInfScanRule.java:353]: Makes sense.

File: addOns/pscanrules/src/main/resources/org/zaproxy/zap/extension/pscanrules/resources/Messages.properties (hunk 2 above)
Comment (thc202) [reply addOns/pscanrules/src/main/resources/org/zaproxy/zap/extension/pscanrules/resources/Messages.properties:218]: +1 to \n.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Comment by psiinon** [comment]
Looks good but now has conflicts

File: addOns/ascanrules/CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ## [61] - 2024-01-24
 ### Changed 
 - Update reference for Server Side Include (Issue 8262) 
+- The Source Code Disclosure - /WEB-INF Folder rule now includes example alert functionality for documentation generation purposes (Issue 6119).
```
Comment (thc202) [inline addOns/ascanrules/CHANGELOG.md:12]: Needs to be moved.

Comment (kingthorin) [reply addOns/ascanrules/CHANGELOG.md:12]: Ugh, thanks. Tackle it in about 30. Thanks for catching it!

**Comment by kingthorin** [comment]
Fixed

**Review by thc202 — APPROVED** [review]

**Review by psiinon — APPROVED** [review]

**Pull request merged**